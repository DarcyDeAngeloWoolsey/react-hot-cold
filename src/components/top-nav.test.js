import React from 'react';
import {shallow, mount} from 'enzyme';
import renderer from 'react-test-renderer';

import TopNav from './top-nav';

describe('<TopNav />', () => {
    it('Renders without crashing', () => {
        shallow(<TopNav />);
    });
        
    it('Renders correctly', () => {
        const tree = renderer.create(
            <TopNav />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
        
    it('Has class clearfix', () => {
        const wrapper = shallow(<TopNav />);
        expect(wrapper.find('ul').hasClass('clearfix')).toEqual(true);
    });
        
    it('Should add a class when the link is clicked', () => {
        const wrapper = shallow(<TopNav />);
//        wrapper.simulate('click');
        expect(wrapper.find('a').first().hasClass('what')).toEqual(true);
    });
        
    it('Should call onInfo when clicked', () => {
        const callback = jest.fn();
        const wrapper = mount(<TopNav onInfo={callback} />);
                                
        wrapper.find('.what').simulate('click')
        
        expect(callback).toHaveBeenCalled();
    })
});