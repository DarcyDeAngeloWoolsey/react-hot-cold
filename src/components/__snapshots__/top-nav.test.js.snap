exports[`<TopNav /> Renders correctly 1`] = `
<nav>
  <ul
    className="clearfix">
    <li>
      <a
        className="what"
        href="#"
        onClick={[Function]}>
        What?
      </a>
    </li>
    <li>
      <a
        className="new"
        href="#"
        onClick={[Function]}>
        + New Game
      </a>
    </li>
  </ul>
</nav>
`;
